### PR TITLE
Make inspecting collections a bit more friendly

### LINKED ISSUE #3308: Calling #inspect on an Event instance doesn't return the content of an ArrayList.
Calling #inspect on a event which contains an ArrayList will return the instance id instead of returning a string of the actual content.

``
"instances"=>#<Java::JavaUtil::ArrayList:0x205196d9>
``

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes #3308 superceeds previous patch

### CODE REVIEW

**Comment by andrewvc** [comment]
@ph let me know how this works for you. I overrode #inspect on java.util.Collection , it seemed more appropriate than just targetting ArrayList.

File: spec/lib/logstash/java_integration_spec.rb [hunk 1]
```diff
@@ -82,6 +82,24 @@
   context "Java::JavaUtil::Collection" do
     subject{Java::JavaUtil::ArrayList.new(initial_array)}
 
+    context "when inspecting" do
+      let(:items) { [:a, {:b => :c}]}
```
Comment (ph) [inline spec/lib/logstash/java_integration_spec.rb]: space after `]`

File: spec/lib/logstash/java_integration_spec.rb [hunk 2]
```diff
@@ -82,6 +82,24 @@
...
+      subject { java.util.ArrayList.new(items)  }
```
Comment (ph) [inline spec/lib/logstash/java_integration_spec.rb]: Extra space before `}`

File: spec/lib/logstash/java_integration_spec.rb [hunk 3]
```diff
@@ -82,6 +82,24 @@
...
+
+      it "should include the contents of the Collection" do
+        #require 'pry'; binding.pry
```
Comment (ph) [inline spec/lib/logstash/java_integration_spec.rb]: We can remove this line ;)

**Comment by ph** [comment]
I have made minor comments, fix them and merge :) LGTM

**Comment by jordansissel** [comment]
Same comments as @ph has, LGTM once those are resolved.

**Pull request closed without merging**